calliope/backend/pyomo/constraints/capacity.py

```diff
@@ -130,8 +132,13 @@ def storage_capacity_constraint_rule(backend_model, loc_tech):
             \\quad \\forall loc::tech \\in loc::techs_{store}
 
     """
-
-    return get_capacity_constraint(backend_model, 'storage_cap', loc_tech)
+    ratio = get_param(backend_model, 'storage_cap_ratio', loc_tech)
+    if po.value(ratio) is not False and po.value(ratio) is not None:
```

Comment:
This might be better placed in its own constraint. At the moment, if a user sets:

```
storage_cap_max: 10
energy_cap_max: 4
storage_cap_ratio: 2
```

`energy_cap_max` will be applied, so `storage_cap` cannot be higher than 8. But, if you set:

```
storage_cap_max: 8
energy_cap_max: 5
storage_cap_ratio: 2
```

`storage_cap_max` is ignored, allowing `storage_cap` to actually go to 10.

Instead, if both the `energy_cap` and `storage_cap` constraints can do their own thing, the constraint in this PR could be an additional constraint which is applied externally, similar to `charge_rate`. To be honest, i'd see it making most sense that `charge_rate` and this constraint are merged into one constraint, in which the link between the two decision variables is either loose (setting a maximum) or strict (as in this constraint)